cdap-ui/app/cdap/components/PreviewData/index.tsx

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { INode, fetchPreview, IRecords, IPreviewData } from 'components/PreviewData/utilities';
+import If from 'components/If';
+import isEmpty from 'lodash/isEmpty';
+import Heading, { HeadingTypes } from 'components/Heading';
+import LoadingSVGCentered from 'components/LoadingSVGCentered';
+import DataTable, { messageTextStyle } from 'components/PreviewData/Table';
+import PreviewTableContainer from 'components/PreviewData/TableContainer';
+import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+import ThemeWrapper from 'components/ThemeWrapper';
+
+const styles = () => {
+  return {
+    messageText: messageTextStyle,
+    headingContainer: {
+      paddingLeft: '10px',
+    },
+  };
+};
+
+interface IPreviewDataViewProps extends WithStyles<typeof styles> {
+  previewId: string;
+  selectedNodeName: string;
+  selectedNode: INode;
+  getStagesAndConnections: () => any;
+  previewStatus: string;
+}
+
+export interface ITableElements {
+  input: any[];
```

Comment:
Could this `input` and `output` be represented in any specific type? Perhaps `IRecords[]`?

Reply:
Good call, types added!